tests/v2/validation/provisioning/rke2/templates_test.go

```diff
@@ -0,0 +1,164 @@
+package rke2
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	v1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/cloudcredentials"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/projects"
+	"github.com/rancher/shepherd/extensions/provisioning"
+	"github.com/rancher/shepherd/extensions/users"
+	password "github.com/rancher/shepherd/extensions/users/passwordgenerator"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/rancher/shepherd/pkg/wait"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	localCluster   = "local"
+	fleetNamespace = "fleet-default"
+)
+
+type ClusterTemplateTest struct {
+	Repo             *v1.ClusterRepo `json:"repo,omitempty" yaml:"repo,omitempty"`
+	TemplateName     string          `json:"templateName,omitempty" yaml:"templateName,omitempty"`
+	TemplateProvider string          `json:"templateProvider,omitempty" yaml:"templateProvider,omitempty"`
+}
+
+type ClusterTemplateTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	session            *session.Session
+	templateConfig     *ClusterTemplateTest
+	cloudCredentials   *cloudcredentials.CloudCredential
+}
+
+func (r *ClusterTemplateTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *ClusterTemplateTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	r.session = testSession
+	r.templateConfig = new(ClusterTemplateTest)
```

Comment:
nit - can these two lines be separated from the `testSession` block? To make it clear they aren't all grouped together.